### PR TITLE
[stable25] chore(CI): Improve rate limit situation when installing phpunit

### PR DESCRIPTION
**Pull request by nickvergessen**
Backport #35957

### CODE REVIEW

**Review by szaimen — APPROVED** [review]

File: .github/workflows/psalm-github.yml [hunk 1]
```diff
@@ -15,7 +15,20 @@ jobs:
...
+        with:
+          php-version: '8.0'
+          extensions: ctype,curl,dom,fileinfo,gd,intl,json,mbstring,openssl,pdo_sqlite,posix,sqlite,xml,zip
+          coverage: none
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Composer install
+        run: composer i
+
```
Comment (come-nc) [inline .github/workflows/psalm-github.yml:31]: This is not consistent with the use of docker://vimeo/psalm-github-actions:4.9.3 below which already has php and psalm

Comment (come-nc) [reply .github/workflows/psalm-github.yml:31]: (but in https://github.com/nextcloud/server/pull/34997 we are merging this file with the other one anyway)

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
ironically, php tests stuck ?

**Comment by skjnldsv** [comment]
```
Invalid workflow file: .github/workflows/lint.yml#L21
The workflow is not valid. .github/workflows/lint.yml (Line: 21, Col: 11): A mapping was not expected .github/workflows/lint.yml (Line: 40, Col: 11): A mapping was not expected
```
https://github.com/nextcloud/server/actions/runs/3839187896

**Comment by skjnldsv** [comment]
Indent is wrong

**Pull request merged**